lib/matplotlib/axes/_axes.py

```diff
@@ -839,30 +844,22 @@ def axline(self, xy1, xy2=None, *, slope=None, **kwargs):
 
             >>> axline((0, 0), (1, 1), linewidth=4, color='r')
         """
-        def _to_points(xy1, xy2, slope):
-            """
-            Check for a valid combination of input parameters and convert
-            to two points, if necessary.
-            """
-            if (xy2 is None and slope is None or
-                    xy2 is not None and slope is not None):
-                raise TypeError(
-                    "Exactly one of 'xy2' and 'slope' must be given")
-            if xy2 is None:
-                x1, y1 = xy1
-                xy2 = (x1, y1 + 1) if np.isinf(slope) else (x1 + 1, y1 + slope)
-            return xy1, xy2
+        if (xy2 is None and slope is None or
+                xy2 is not None and slope is not None):
+            raise TypeError(
+                "Exactly one of 'xy2' and 'slope' must be given")
```

Comment:
This check should be done in `_AxLine.__init__` to be sure that class is consistent within itself. We may or may not repeat the check here.

Reply:
Right, makes sense. Done in da0b3449512d60096d7891e3846f5c3495572d4a.